src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -348,8 +348,14 @@ public HttpResponse doFinishLogin(final StaplerRequest request, final StaplerRes
         return HttpResponses.redirectTo(redirectUrl);
     }
 
-    private String getEffectiveLogoutUrl() throws NullPointerException {
-        return StringUtils.isNotBlank(getLogoutUrl()) ? getLogoutUrl() : Jenkins.getInstanceOrNull().getRootUrl() + SamlLogoutAction.POST_LOGOUT_URL;
+    private String getEffectiveLogoutUrl() {
+        Jenkins j = Jenkins.getInstanceOrNull();
+        if(j != null) {
+            return StringUtils.isNotBlank(getLogoutUrl()) ? getLogoutUrl() : j.getRootUrl() + SamlLogoutAction.POST_LOGOUT_URL;
+        }
+        else {
+            return null;
```

Comment:
this change the behavior of the method, the previous implementation does not return null in any case